File: Blind75/Arrays&Hashing/Product_of_Array_Except_Self.py
# Leetcode 238
from typing import List
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1]
        for val in nums[:-1]:
            res.append(res[-1]*val)

        mul=1
        for idx in range(len(nums)-1,-1,-1):
            res[idx] *= mul
            mul*=nums[idx]
        
        return res

File: Blind75/Arrays&Hashing/Product_of_Array_Except_Self.java
/*
 * Leetcode 238
 */

class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] result = new int[nums.length];
        int mul = 1;
        for(int i=0;i<nums.length;i++){
            result[i] = mul;
            mul*=nums[i];
        }
        mul = 1;
        for(int i=nums.length-1;i>=0;i--){
            result[i]*=mul;
            mul*=nums[i];
        }
        return result;
    }
}